internal/kafka/internal/workers/cluster_mgrs/cleanup_clusters_mgr.go

```diff
@@ -95,6 +101,20 @@ func (m *CleanupClustersManager) processCleanupClusters() error {
 	return errList
 }
 
+func (m *CleanupClustersManager) reconcileCleanupEnterpriseCluster(cluster api.Cluster) error {
+	glog.Infof("Removing Enterprise Dataplane cluster %s fleetshard service account", cluster.ClusterID)
+	serviceAccountRemovalErr := m.kasFleetshardOperatorAddon.RemoveServiceAccount(cluster)
+	if serviceAccountRemovalErr != nil {
+		return errors.Wrapf(serviceAccountRemovalErr, "failed to removed Dataplane cluster %s fleetshard service account", cluster.ClusterID)
+	}
+	glog.Infof("Soft deleting the Enterprise Dataplane cluster %s from the database", cluster.ClusterID)
+	deleteError := m.clusterService.DeleteByClusterID(cluster.ClusterID)
+	if deleteError != nil {
+		return errors.Wrapf(deleteError, "failed to soft delete Dataplane cluster %s from the database", cluster.ClusterID)
+	}
+	return nil
```

Comment:
could this be simplified to `m.clusterService.DeleteByClusterID(cluster.ClusterID)` or whatever is gonna be placed here (e.g. hard delete cluster function)?

Reply:
unfortunately no, because the method returns a simple error, while the clusterService returns a service error, which is not quite nil even when there is no error. See @ziccardi comment below on the error vs. service error pains.